### DISCUSSION TITLE
Is it possible to remove the top bar?

### DISCUSSION THREAD

**Discussion by vinoff** [ghostty-org/ghostty#3158] [Q&A] [answered]
Two questions:

1) Is it possible to remove the top bar (title bar?)? The one above the tab bar?

2) Is it possible to change the width/height/size of the tab bar?

EDIT: I am on Linux.

**Comment by vinoff** [comment]
gtk-titlebar = false fixed 1.

**Reply by iceener** [reply]
Doesn't seem to work for me. Is that the only setting?

**Reply by ghost** [reply]
> Doesn't seem to work for me. Is that the only setting?

```
window-decoration = false
```

Quote from docs, if set to false:
```md
windows won't have native decorations, i.e. titlebar and borders. On macOS this also disables tabs and tab overview. 
```

If in mac, this is better:
```
macos-titlebar-style = hidden
```

**Reply by knownasnaffy** [reply]
`gtk-titlebar=false` swaps the gtk titlebar with the one that your current DE offers.

**Comment by ecdye** [comment]
For number 2 I don't believe so. We use the native UI tools so it is probably configured by that.

**Comment by ghost** [comment]
```
window-decoration = false
```

**Comment by ghost** [comment]
> Two questions:
> 
>     1. Is it possible to remove the top bar (title bar?)? The one above the tab bar?
> 
>     2. Is it possible to change the width/height/size of the tab bar?
> 
> 
> EDIT: I am on Linux.

Currently No. But, tabs location can be changed :
```
gtk-tabs-location

Determines the side of the screen that the GTK tab bar will stick to. Top, bottom, left, right, and hidden are supported. The default is top.

If this option has value left or right when using Adwaita, it falls back to top. hidden, meaning that tabs don't exist, is not supported without using Adwaita, falling back to top.

When hidden is set and Adwaita is enabled, a tab button displaying the number of tabs will appear in the title bar. It has the ability to open a tab overview for displaying tabs. Alternatively, you can use the toggle_tab_overview action in a keybind if your window doesn't have a title bar, or you can switch tabs with keybinds.
```

This is from docs.

**Comment by ghost** [comment]
For number 2. you can edit `gkt.css` file (on linux it is located at `$XDG_CONFIG_HOME/gtk-4.0/gtk.css`). I think that the element you are targeting is called 'tab', but you should refer to the gtk documentation for more details.

Example (makes tabs comically large):

```css
tab {
  padding: 2rem;
}
```

You can also use `GTK_DEBUG=interactive` to start ghostty with gtk inspector attached to it; it can help you locate the elements you want to style. Also, AFAIK gkt's css is not the standard css, it is fairly limited.

**Reply by vinoff** [reply]
I did that but they are still enormous. Sadly.

**Comment by elitistphoenix** [comment]
I'd like to know or request the option too please. Currently just seems like wasted space.

![Screenshot_2025-01-10_11-35-16](https://github.com/user-attachments/assets/173144c7-f214-48ef-92b1-a4eccdca7109)

**Comment by fvnky07** [comment]
any updates? because the title bar is really useless. should have the option to disable it. surely it would be an easy pr?

**Reply by rhodes-b** [reply]
its `gtk-titlebar = false`
please read?

if you dont want big tabs
https://ghostty.org/docs/config/reference#gtk-adwaita

**Reply by fvnky07** [reply]
>gtk-titlebar=false swaps the gtk titlebar with the one that your current DE offers.

**Reply by rhodes-b** [reply]
> >gtk-titlebar=false swaps the gtk titlebar with the one that your current DE offers.
> 

In 1.0 it did it shouldn't in 1.0.1 or 1.1.0
Try 1.1.0 make sure to look at the new options for window decorations

**Comment by elitistphoenix** [comment]
![image](https://github.com/user-attachments/assets/4ae9b0d3-d7d5-4cc2-ad5b-5e4d83881a95)
![image](https://github.com/user-attachments/assets/a95ecc76-9d24-4124-a04d-56b432a194a3)
v1.0.1

**Comment by gwirn** [comment]
`gtk-tabs-location` should hide the tab bar (v1.1.0)

**Comment by pluiedev** [comment, marked as answer]
> 1. Is it possible to remove the top bar (title bar?)? The one above the tab bar?

Yes. Set the `window-decoration` setting to either:
  - `server` if you want to use your system's title bar. Supported since 1.1 for Wayland, or 1.1.1 for X11 (#5533);
  - `none` if you don't want a title bar at all — broken on GNOME on 1.1, fixed in 1.1.1 (#5463). 

> 2. Is it possible to change the width/height/size of the tab bar?

[Custom CSS](https://ghostty.org/docs/config/reference#gtk-custom-css).

**Reply by vinoff** [reply]
And what exactly should we add to the Custom CSS file for the tab bar to be smaller? I tried height, padding, margins, font-size, etc. Nothing made it smaller.

**Reply by mschubert** [reply]
To add to this: If you only want to remove the top bar when the window is maximized, use:

```sh
# ~/.config/ghostty/config
gtk-titlebar-hide-when-maximized = true
```

Discussed in https://github.com/ghostty-org/ghostty/issues/3381, commit is here: https://github.com/rockorager/ghostty/commit/46684f29dd89ba3214e10c48e5a97f56df84b85b

**Reply by mia-duebel** [reply]
Additionally, there's
```ini
# ~/.config/ghostty/config
gtk-titlebar = false
```
